service/storage_service.cc

```diff
@@ -2089,14 +2107,20 @@ future<> storage_service::raft_replace(raft::server& raft_server, raft::server_i
 
     // add myself to topology with request to replace
     while (!_topology_state_machine._topology.contains(raft_server.id())) {
-        slogger.info("raft topology: adding myself to topology for replace: {} replacing {}", raft_server.id(), replaced_id);
+        slogger.info("raft topology: adding myself to topology for replace: {} replacing {}, ignored nodes: {}", raft_server.id(), replaced_id, ignored_ids);
         auto guard = co_await _group0->client().start_operation(&_abort_source);
 
         auto it = _topology_state_machine._topology.normal_nodes.find(replaced_id);
         if (it == _topology_state_machine._topology.normal_nodes.end()) {
             throw std::runtime_error(::format("Cannot replace node {}/{} because it is not in the 'normal' state", replaced_ip, replaced_id));
         }
 
+        for (const auto& ignored_id : ignored_ids) {
+            if (!_topology_state_machine._topology.find(ignored_id)) {
+                throw std::runtime_error(::format("raft topology replace: ignored node {} is not found in the cluster", ignored_id));
```

Comment:
> Is this the same behavior as for old replace?

I think so. The old replace also throws `runtime_error` if an ignored node is not in the cluster (see `storage_service::parse_node_list`). Only the error message is different.

> BTW let's remove "raft topology " prefix from errors that we throw through user APIs ("raft topology" is an implementation detail). It's fine to keep it in log messages.

I changed the errors added by me. What about the ones already present in `raft_removenode`? Should I fix them in another PR, since they are unrelated?

Reply:
> Should I fix them in another PR, since they are unrelated?

Sure, after this PR is merged.